app/models/mixins/ansible_playbook_mixin.rb

```diff
@@ -33,11 +33,13 @@ def hosts_array(hosts_string)
 
   def playbook_log_stdout(log_option, job)
     return unless %(on_error always).include?(log_option)
-    return if log_option == 'on_error' && job.raw_status.succeeded?
+    return if log_option == 'on_error' && job && job.raw_status.succeeded?
 
-    $log.info("Stdout from ansible job #{job.name}: #{job.raw_stdout('txt_download')}")
+    $log.info("Stdout from ansible job #{job.name}: #{job.raw_stdout('txt_download')}") if job
   rescue StandardError => err
-    $log.error("Failed to get stdout from ansible job #{job.name}")
+    msg = "Failed to get stdout from ansible job"
+    msg << " #{job.name}" if job
```

Comment:
A few things here:

- Would it make sense to rescue `NoMethodError` above here as well?  We could avoid a few of the `if` calls above by doing that
- To address @d-m-u 's [concern](https://github.com/ManageIQ/manageiq/pull/20738/files#r512661330), could we change the code here to this:

```ruby
    msg = "Failed to get stdout from ansible job #{job ? job.name : '<UknownJob>'}"
```

* * *

That said, I think he concern raised by @chessbyte ([here](https://github.com/ManageIQ/manageiq/pull/20738/files#r512711874)) and @Fryguy ([here](https://github.com/ManageIQ/manageiq/pull/20738#issuecomment-717289680)) are valid, so that might make the above suggesting unnecessary if we determine the root cause for this should normally not happen (I am not sure myself to be honest).